File: movieApp/app/src/main/java/com/lbc/practice/movieapp/di/AppComponent.java
//package com.lbc.practice.movieapp.di;
//
//
//import android.app.Application;
//
//import com.lbc.practice.movieapp.MovieApplication;
//import com.lbc.practice.movieapp.view.favorite.FavoriteModule;
//import com.lbc.practice.movieapp.view.main.MainModule;
//import com.lbc.practice.movieapp.view.second.SecondModule;
//
//import javax.inject.Singleton;
//
//import dagger.BindsInstance;
//import dagger.Component;
//import dagger.android.AndroidInjector;
//
//@Singleton
//@Component(modules = {})
//public interface AppComponent extends AndroidInjector<MovieApplication> {
//
//    @Component.Builder
//    interface Builder {
//
//        @BindsInstance
//        AppComponent.Builder application(Application application);
//
//        AppComponent build();
//
//    }
//}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/di/AppComponent.kt
package com.lbc.practice.movieapp.di;


import android.app.Application;

import com.lbc.practice.movieapp.MovieApplication;
import com.lbc.practice.movieapp.data.resource.MovieRepoistoryModule
import com.lbc.practice.movieapp.data.resource.remote.RemoteModule

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;


@Singleton
@Component(modules = arrayOf(RemoteModule::class, MovieRepoistoryModule::class, ApplicationModule::class,
        ActivityBindingModule::class, AndroidSupportInjectionModule::class))
interface AppComponent : AndroidInjector<MovieApplication> {


    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): AppComponent.Builder
        fun build(): AppComponent
    }
}